modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1275,6 +1275,13 @@ CV__DNN_INLINE_NS_BEGIN
          CV_WRAP void detect(InputArray frame, CV_OUT std::vector<int>& classIds,
                              CV_OUT std::vector<float>& confidences, CV_OUT std::vector<Rect>& boxes,
                              float confThreshold = 0.5f, float nmsThreshold = 0.0f);
+         CV_WRAP Mat blobFromImages(InputArray images_, double scalefactor = 1.0,
+             Size size = Size(), const Scalar& mean = Scalar(),
+             bool swapRB = false, Rect crop = Rect(), int ddepth = CV_32F);
```

Comment:
That should be a static method.

Reply:
@dkurt do you mean  I should keep the function definition as this 
```static Mat blobFromImages(InputArray images_, double scalefactor=1.0, Size size = Size() ,const Scalar& mean = Scalar(), bool swapRB=false, const Rect& crop=Rect(),int ddepth=CV_32F)``` is this the reason why the build is failing

Reply:
I mean it should work in the same way as existing `blobFromImage` so defined as `cv::dnn::blobFromImage`, no in `DetectionModel`. There are three different issues actually:

1) blobFromImage with cv::Rect
2) correct detection outputs from DetectionModel considering crop=True
3) somehow retrieve the ROI position which is determined inside blobFromImage with crop=True

So first, I'd like to recommend to take a look at (2) issue to fix object detection with crop=True.

Reply:
1) This can be done by declaring the same blobFromImage in cv::dnn namespace with Rect as one of the arguments.
2) When crop= True in blobFromImage function the images in the blob have only its central part but how is it related to the detection outputs of the DetectionModel.

Reply:
@dkurt can you guide on how to approach the second issue

Reply:
@themechanicalcoder, in example, try to run some of the detectors with `crop=True`. The detections must be wrong. The problem is that detected objects should be inside cropped image but at the end we still multiply by full image sizes (width and height).

Reply:
@dkurt I think the point you mentioned is taken care of by [this line](https://github.com/opencv/opencv/blob/bf2f7b0f8b7c4687f22610a62daea603ded9345d/modules/dnn/src/model.cpp#L257) and it is multiplied by the ```size``` of the cropped image only.
According to me, the problem is that the coordinates of the bbox needs to be shifted by the [point](https://github.com/opencv/opencv/blob/bf2f7b0f8b7c4687f22610a62daea603ded9345d/modules/dnn/src/dnn.cpp#L339) ```Point(0.5 * (images[i].cols - size.width), 0.5 * (images[i].rows - size.height)```  so that we can obtain the original coordinates of the bbox in the image.

Reply:
It depends on how image has been cropped, actually. Please check

https://github.com/opencv/opencv/blob/3d14dd4e3953aa7b7e2ab3a388ca7bf08fe99d99/modules/dnn/src/dnn.cpp#L334-L345

Reply:
The dimensions of the bbox computed are according to the size of the cropped image so multiplying the height and width by 1/resizeFactor should give the dimensions of the actual bbox